util/limbo_task_db.cpp

```diff
@@ -95,9 +95,9 @@ void LimboTaskDB::scan_user_tasks() {
 		tasks_cache[LimboTaskDB::get_misc_category()] = List<String>();
 	}
 
-	for (int i = 1; i < 4; i++) {
-		String dir1 = ProjectSettings::get_singleton()->get_setting_with_override("limbo_ai/behavior_tree/user_task_dir_" + itos(i));
-		_populate_from_user_dir(dir1, &tasks_cache);
+	PackedStringArray user_task_directories = GLOBAL_GET("limbo_ai/behavior_tree/user_task_dir");
+	for String user_task_dir : user_task_directories) {
```

Comment:
```suggestion
	for (const String &user_task_dir : user_task_directories) {
```
Compilation error here. Also, it's better to get strings by reference, otherwise it allocates unnecessary objects.

Reply:
Looking back, i am actually a little surprised. for context, I haven't really figured out a good setup to work with scons in visual studio code (on windows), as it thinks most of the code is inaccessible because of the preprocessor conditions, despite them getting set when compiling from the command line. The thing that baffles me a little now is that MSVC is actually fine with this line of code and compiled it into a working version of the editor.

it is, however, very wrong.